tests/TestCase/Policy/RequestPolicyTest.php

```diff
@@ -0,0 +1,328 @@
+                    'id' => 1,
+                    'roles' => ['admin'],
+                ],
+            ],
+            'ok using simple class name for custom policy' => [
+                true,
+                (new ServerRequest(['uri' => new Uri('/dashboard/profile')]))
+                    ->withParam('controller', 'Dashboard')
+                    ->withParam('action', 'profile'),
+                [
+                    'rules' => [
+                        'Dashboard' => 'CustomPolicy',
+                    ],
+                ],
+                [
+                    'id' => 1,
+                    'roles' => ['admin'],
+                ],
+            ],
+            'ko using simple class name for custom policy' => [
+                new MissingMethodException(['canAccess', 'access', InvokeCustomPolicy::class]),
+                (new ServerRequest(['uri' => new Uri('/dashboard/profile')]))
+                    ->withParam('controller', 'Dashboard')
+                    ->withParam('action', 'profile'),
+                [
+                    'rules' => [
+                        'Dashboard' => 'InvokeCustomPolicy',
+                    ],
+                ],
+                [
+                    'id' => 1,
+                    'roles' => ['admin'],
+                ],
+            ],
+            'ko invalid policy instance' => [
+                new \LogicException('Invalid rule for Dashboard::profile() in RequestPolicy'),
+                (new ServerRequest(['uri' => new Uri('/dashboard/profile')]))
+                    ->withParam('controller', 'Dashboard')
+                    ->withParam('action', 'profile'),
+                [
+                    'rules' => [
+                        'Dashboard' => [
+                            'profile' => new \ArrayObject(),
+                        ],
+                    ],
+                ],
+                [
+                    'id' => 1,
+                    'roles' => ['admin'],
+                ],
+            ],
+            'ko invalid rule' => [
+                new \LogicException('Invalid Rule for Dashboard in RequestPolicy'),
+                (new ServerRequest(['uri' => new Uri('/dashboard/profile')]))
+                    ->withParam('controller', 'Dashboard')
+                    ->withParam('action', 'profile'),
+                [
+                    'rules' => [
+                        'Dashboard' => new \ArrayObject(),
+                    ],
+                ],
+                [
+                    'id' => 1,
+                    'roles' => ['admin'],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Test that rules work as expected.
+     *
+     * @param mixed $expected The expected result
+     * @param \Cake\Http\ServerRequest $request The request
+     * @param array $policyConfig Policy configuration
+     * @param array $identityData Identity data
+     * @return void
+     * @dataProvider canAccessProvider()
+     */
```

Comment:
missing phpdoc `@covers ::canAccess()`

Reply:
SInce the test cover all class methods I'm going to add `@covers` to class doc block